actions/check-thirdparty-notice/action.yml

```diff
@@ -0,0 +1,84 @@
+# SPDX-FileCopyrightText: Copyright (C) 2025 Opal Health Informatics Group at the Research Institute of the McGill University Health Centre <john.kildea@mcgill.ca>
+#
+# SPDX-License-Identifier: TBD
+
+# Ensures that declared project dependencies and the third-party notice in THIRDPARTY.md are in sync.
+# I.e., each dependency has to have a corresponding heading in the notice.
+# Warns about dependencies missing as well as dependencies that are declared in addition.
+#
+# Note: Currently supports npm, composer and pip.
+
+name: Check Thirdparty Notice
+description: Ensures that declared project dependencies and the third-party notice in THIRDPARTY.md are in sync.
+
+inputs:
+  package-file:
+    description: The filename of an npm package.json file (can be relative to the current working directory)
+    default: ""
+  composer-file:
+    description: The filename of a composer.json file (can be relative to the current working directory)
+    default: ""
+  pip-requirements-files:
+    description: |
+      The filename(s) of a requirements file (can be relative to the current working directory).
+      If specifying more than one file, separate with spaces.
+    default: ""
+  additional-dependencies:
+    description: |
+      Additional dependencies that are allowed to be listed in the third-party notice.
+      If specifying more than one package, separate with spaces.
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    # Setup
+    - name: Clone the caller's repo to access dependency files and the third-party notice
+      uses: actions/checkout@v4.2.2
+      with:
+        persist-credentials: false
+    - name: Clone check_thirdparty_notice.py from the actions repo
+      uses: actions/checkout@v4.2.2
+      with:
+        repository: opalmedapps/actions
+        path: opalmedapps-actions
+        # See: https://github.com/actions/checkout?tab=readme-ov-file#fetch-only-a-single-file
+        sparse-checkout: |
+          actions/check-thirdparty-notice/check_thirdparty_notice.py
+        sparse-checkout-cone-mode: false
+        persist-credentials: false
```

Comment:
@mschoettle Is this fine, or do you prefer I change it to https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action?platform=mac#creating-an-action-metadata-file ?

Reply:
Weird. Maybe because it is not marked as an executable script. But copying over is also fine.